libcriu: add API for join-ns option

rst0git opened the pull request:
We use the join-ns RPC API in runc to enable checkpoint/restore of containers with shared namespaces. Shared namespaces are often used when containers run inside Kubernetes Pod. However, crun is using libcriu to interface with CRIU and libcriu currently doesn't provide an API for join-ns. This pull request adds the libcriu API necessary to enable checkpoint/restore of containers with shared namespaces with crun.

adrianreber commented:
I was a bit confused about the interface. Path names instead of file descriptors. Looking at the RPC and CLI interface I think your implementation is correct as RPC and CLI also use path names and not file descriptors. Which is confusing because inherit-fd uses file descriptors.

Everything looks good. Unfortunately the arm tests are currently not running. I will wait for the results from those tests and then it looks ready to be merged.

rst0git commented:
Thank you Adrian for the code review. I was hoping to ask for feedback for the test case of join-ns. It would currently check that checkpoint/restore has succeeded. While this is sufficient to check if the join-ns option has been passed to CRIU (example below), we don't currently have tests to verify that `--join-ns` actually works. Perhaps we could add the tests in runc/crun?
```bash
sudo make -C test/others/libcriu/ run
sudo cat test/others/libcriu/wdir/i/test_join_ns/restore.log | grep 'join namespace'
(00.000244) Added ipc:/proc/66302/ns/ipc join namespace
(00.000264) Added uts:/proc/66302/ns/uts join namespace
(00.000281) Added time:/proc/66302/ns/time join namespace
```
@avagin do you have any advice?

adrianreber commented:
> Thank you Adrian for the code review. I was hoping to ask for feedback for the test case of join-ns. It would currently check that checkpoint/restore has succeeded. While this is sufficient to check if the join-ns option has been passed to CRIU (example below), we don't currently have tests to verify that `--join-ns` actually works. Perhaps we could add the tests in runc/crun?
> 
> ```shell
> sudo make -C test/others/libcriu/ run
> sudo cat test/others/libcriu/wdir/i/test_join_ns/restore.log | grep 'join namespace'
> (00.000244) Added ipc:/proc/66302/ns/ipc join namespace
> (00.000264) Added uts:/proc/66302/ns/uts join namespace
> (00.000281) Added time:/proc/66302/ns/time join namespace
> ```
> 
> @avagin do you have any advice?

I would compare `/proc/<PID>/ns/<namespace>` just as we do in other tests.

rst0git commented:
> I would compare `/proc/<PID>/ns/<namespace>` just as we do in other tests.

Thanks, I've updated the test case.

adrianreber commented:
I see that you changed `run.sh`. Could you add it to the shellcheck test targets?
